app/code/Magento/InventorySales/Model/ResourceModel/ReplaceSalesChannelsDataForStock.php

```diff
@@ -53,7 +53,7 @@ public function execute(array $salesChannels, int $stockId)
                     CreateSalesChannelTable::STOCK_ID => $stockId,
                 ];
             }
-            $connection->insertMultiple($tableName, $salesChannelsToInsert);
+            $connection->insertOnDuplicate($tableName, $salesChannelsToInsert);
```

Comment:
In this methid we have next logic:
1) Delete all with string `$connection->delete($tableName, [CreateSalesChannelTable::STOCK_ID . ' = ?' => $stockId]);`
2) Create new with `$connection->insertMultiple($tableName, $salesChannelsToInsert);`

So need to investigate why do we have some data after the first step
Looks like we have duplication of sales channels in $salesChannels list
Need to investigate and fix this problem on higher level